### PR TITLE
Run coverage report for PRs

### LINKED ISSUE #3591: Run coverage report for PRs
## Feature Request

Previously we disabled running coverage in PR #3247 because tests are run twice which slows down CI a lot.

Now [Tarpaulin supports returning test error codes](https://github.com/xd009642/tarpaulin/issues/119) so maybe we could merge them together and bring Tarpaulin back for PRs.

### PR DESCRIPTION
**Pull request by bb7133**
This PR will close #3591

Signed-off-by: bb7133 <bb7133@gmail.com>

## What have you changed? (mandatory)

Enable code coverage report in CI

## What are the type of the changes? (mandatory)

A simple feature

## How has this PR been tested? (mandatory)

There's no test related

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

https://github.com/tikv/tikv/pull/3247
https://github.com/tikv/tikv/issues/3591

### CODE REVIEW

**Comment by siddontang** [comment]
Thank you very much @bb7133 

PTAL @breeswish @Hoverbear

File: .circleci/config.yml [hunk 1]
```diff
@@ -129,16 +127,10 @@ jobs:
           paths:
             - ~/tikv_cache_build
       - run:
-          name: Testing
-          command: make trace_test
```
Comment (ice1000) [inline .circleci/config.yml]: Why remove `make trace_test`?

**Comment by breeswish** [comment]
Oh, I think this PR is blocked by https://github.com/tikv/tikv/issues/3623 . Let's hold it until #3623 is fixed.. otherwise all PR's CI will fail as well

File: .circleci/config.yml (hunk 1 above)
Comment (breeswish) [reply .circleci/config.yml]: See #3591

**Comment by bb7133** [comment]
OK, thanks for the comments! @siddontang @breeswish @ice1000

**Comment by Connor1996** [comment]
This is the PR https://github.com/tikv/tikv/pull/3719 to fix #3623. Once it is merged, this PR can work as expected. 😉

**Comment by Connor1996** [comment]
Seems that it can be merged now. PTAL @Hoverbear @breeswish

**Comment by breeswish** [comment]
Hmm. Seems that master is still failing https://circleci.com/gh/tikv/tikv/tree/master

**Comment by Connor1996** [comment]
@breeswish it is weird, when I tested other cases seem stable enough that they doesn't panic for several times. The only difference is using a newer version of tarpaulin, maybe there is some performance improvement. And as this PR shown, all tests are passed. So we can rerun CI in this PR for several times, if okay, IMO, it can be merged.

**Comment by breeswish** [comment]
Here is a list of blocking issues, which is unstable tests in coverage reports:

- [ ] https://github.com/tikv/tikv/issues/2948
- [ ] https://github.com/tikv/tikv/issues/3162
- [x] https://github.com/tikv/tikv/issues/3669
- [x] https://github.com/tikv/tikv/pull/3719
- [x] https://github.com/tikv/tikv/issues/3793
- [ ] https://github.com/tikv/tikv/issues/3804
- [ ] https://github.com/tikv/tikv/issues/3829
- [ ] https://github.com/tikv/tikv/issues/3830
- [ ] https://github.com/tikv/tikv/issues/3831

**Comment by breeswish** [comment]
We decided to migrate our CI infrastructure for PRs to own Jenkins CI, so that we can have more control and make it faster. Closing.

**Comment by bb7133** [comment]
Yeah, just heard some details from TiDB meetup~

**Pull request closed without merging**